sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -50,17 +50,19 @@ case class Mode(
   override def inputTypes: Seq[AbstractDataType] = Seq(AnyDataType)
 
   override def checkInputDataTypes(): TypeCheckResult = {
-    if (UnsafeRowUtils.isBinaryStable(child.dataType) || child.dataType.isInstanceOf[StringType]) {
+    // TODO: SPARK-48700: Mode expression for map type with collated fields
```

Comment:
```suggestion
    // TODO: SPARK-49358: Mode expression for map type with collated fields
```

Reply:
`SPARK-48700` is the current ticket

I created a new ticket for MapType: https://issues.apache.org/jira/browse/SPARK-49358